backend/main.py

```diff
@@ -11,21 +13,44 @@
 origins = os.environ.get("ALLOW_CORS", "").split(",")
 
 app.add_middleware(
-    CORSMiddleware,
-    allow_origins=origins,
-    allow_credentials=True,
-    allow_methods=["*"],
-    allow_headers=["*"],
+  CORSMiddleware,
+  allow_origins=origins,
+  allow_credentials=True,
+  allow_methods=["*"],
+  allow_headers=["*"],
 )
 
-@app.get('/')
-async def root():
-  return {"message": "Hello World"}
+class BobaOptionMilkType(graphene.ObjectType):
+    name = graphene.String()
+
+class BobaOptionToppingsType(graphene.ObjectType):
+    name = graphene.String()
+    price = graphene.Float()
+
+class BobaOptionType(graphene.ObjectType):
+    milk = graphene.Field(lambda: BobaOptionMilkType)
+    toppings = graphene.List(lambda: BobaOptionToppingsType)
+
+
+class BobaType(graphene.ObjectType):
+    id = graphene.ID()
+    name = graphene.String()
+    description = graphene.String()
+    price = graphene.Float()
+    imageLink = graphene.String()
+    options = graphene.Field(lambda: BobaOptionType)
+
+class BobaQuery(graphene.ObjectType):
+    bobaList = graphene.List(BobaType)
+
+    def resolve_bobaList(self, info):
+      with open('./database/bobalist.json') as f:
+          bobajson = json.load(f)['data']
+      return bobajson
 
 @app.get('/boba_list')
 async def get_boba_list():
   with open('./database/bobalist.json') as f:
-      # Load the JSON data into a Python object
     bobaList = json.load(f)
   return bobaList["data"]

```

Comment:
I know you added GraphQL for `boba_list` but does your front end still work after you remove this `@app.get('/boba_list')`

Reply:
i thought about this and deleted before but put it back because i am not sure whether i would need it, i will update on next push.

Reply:
Keep for now, but remove next PR

Reply:
okay, will removed in next pr 
the rest been updated on d55f7b42f7cff17fde66ffdc9abc25be6f1721ae